File: easy/python/0122.py
class Solution:
    def maxProfit(self, prices: 'List[int]') -> 'int':
        if not prices:
            return 0
        res = 0
        for i in range(len(prices)-1):
            cur = prices[i]
            next_ = prices[i+1]
            if next_ > cur:
                res += next_ - cur
        return res

File: easy/cpp/0122.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int min = prices[0];
        int ret = 0;
        for(int i = 1 ; i < prices.size(); i++){
            if(prices[i] - min > 0){
                ret += prices[i] - min;
                min = prices[i];
            } 
            if(prices[i] < min) min = prices[i];
        }
        return ret;
    }
};